src/handle_trailer.sh

```diff
@@ -0,0 +1,169 @@
+include "${KW_LIB_DIR}/lib/kw_config_loader.sh"
+include "${KW_LIB_DIR}/lib/kwlib.sh"
+
+declare -gA options_values
+
+# This fnuction performs operations over trailers in
+# either patches or commits. It checks if given argument
+# is a valid commit reference or patch path and uses the
+# correct command to perform the task.
+# If that's not the case, a warning message will tell
+# the user this argument was ignored.
+#
+# Also, if no operation option is given, then an error message
+# followed by a helper message is printed to the user.
+#
+# @PATCH_OR_SHA Holds a patch path or commit reference.
+# @TRAILER_TAG Holds the key of the operation's trailer.
+# @TRAILER_VALUE Holds the value of the operation's trailer.
+# @flag Defines the type of output this function will have.
+# @cmd Holds the command used to perform the trailer operation.
+function handle_trailer_main()
+{
+  local PATCH_OR_SHA
+  local TRAILER_TAG
+  local TRAILER_VALUE
+  local flag
+  local cmd
+
+  if [[ "$1" =~ -h|--help ]]; then
+    handle_trailer_help "$1"
+    exit 0
+  fi
+
+  parse_handle_trailer_options "$@"
+  if [[ "$?" -gt 0 ]]; then
+    complain "${options_values['ERROR']}"
+    return 22 # EINVAL
+  fi
+
+  [[ -n "${options_values['VERBOSE']}" ]] && flag='VERBOSE'
+  flag=${flag:-'SILENT'}
+
+  read -ra PATCH_OR_SHA <<< "${options_values['PATCH_OR_SHA']}"
+  TRAILER_TAG=${options_values['TRAILER_TAG']}
+  TRAILER_VALUE=${options_values['TRAILER_VALUE']}
+
+  if [[ -z "$TRAILER_TAG" ]]; then
+    complain 'An option is required to use this command.'
+    handle_trailer_help
+    return 22 # EINVAL
+  fi
+
+  for arg in "${PATCH_OR_SHA[@]}"; do
+    # Check if given argument is either a patch or valid commit reference,
+    # then build the correct command.
+    if [[ $(git cat-file -t "$arg" 2> /dev/null) == 'commit' ]]; then
+      cmd="git commit --quiet --amend --no-edit --trailer \"${TRAILER_TAG} ${TRAILER_VALUE}\""
```

Comment:
I'm afraid this can't be done, at least we don't know how to build this command correctly without `\"`. The problem with using single-quotes is that we have multiple nested strings. Replacing the `\"` for `'` messes up the cmd's building process and generates commands that are not correctly executed.

We've tried to think of ways to build the command without it, but it messes up the format.